packages/pingcap-ui/src/utils/nav.utils.js

```diff
@@ -0,0 +1,23 @@
+export function replaceNavLinks(items, urlPrefixRegexp) {
+  if (!Array.isArray(items)) {
+    return items;
+  }
+
+  const newItems = [];
+
+  for (const item of items) {
+    const newItem = { ...item };
+
+    if (item.items) {
+      newItem.items = replaceNavLinks(item.items, urlPrefixRegexp);
+    }
+
+    if (typeof item.link === 'string') {
```

Comment:
RamdaJS is used as the default dependency, so we could detect string type by Ramda helper like:

```javascript
R.is(String, item.link)
```

https://ramdajs.com/docs/#is